# First Steps with Python and Jupyter 

![](https://i.imgur.com/gvSnw4A.png)




This tutorial covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter. 
we perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the '+' ,'-','/','//', '%', and '**' operators are used for?

As you might expect, operators like / and * take precedence over other operators like + and - as per mathematical conventions. You can use parentheses, i.e. ( and ), to specify the order in which operations are performed.

In [2]:
# addition(+)operator
a=10
b=20
c=a+b
print(c)

30

In [4]:
x=2
y=4
print(x+y)

6


In [6]:
# subtract (-)operator
a=10
b=20
c=a-b
print(c)


-10


In [7]:
a=10
b=20
c=b-a
print(c)

10


In [10]:
x=3
y=2
print(x-y)

1


In [11]:
#Multiplication(*)operator
x1=8
y1=12
z1=x1*y1
print(z1)

96


In [15]:
# Now we make table of 2
x=2
for i in range(11):
    print(x,'*',i,'=',x*i)

2 * 0 = 0
2 * 1 = 2
2 * 2 = 4
2 * 3 = 6
2 * 4 = 8
2 * 5 = 10
2 * 6 = 12
2 * 7 = 14
2 * 8 = 16
2 * 9 = 18
2 * 10 = 20


In [16]:
a1=2
b1=4
a1*b1

8

In [17]:
# Division (/) operator
100/7

14.285714285714286

In [18]:
a=56
b=12
c=a/b
print(c)

4.666666666666667


In [19]:
print(4/2)
print(4/4)

2.0
1.0


In [20]:
# Floor Division(//) operator
100//7


14

In [21]:
a//b

4

In [23]:
4//2

2

In [26]:
x=10
y=20
z=y//x
print(z)

2


In [30]:
# Modulus and remainder(%) operation 
x=4
y=9
z=y%x
print(z)

1


In [31]:
21%2

1

In [32]:
23%3

2

In [33]:
# Exponent (**)operator 
3**2

9

In [34]:
x=5
x**5

3125

In [35]:
x=2
2**2

4

In [36]:
3**2

9

In [37]:
4**2

16

In [38]:
5**2

25

In [39]:
6**2

36

In [40]:
5**3

125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

You can use the empty cells below and add more cells if required.

Question 1:-The population of a town is 198568. Out of them 45312 are men and 35678 are women. Find the number of children in the town.

In [42]:
total_popul=198568
total_male=45312
total_female=35678
sum1=total_male + total_female
total_children=total_popul - sum1
print(total_children)

117578


Question 2:- A shopkeeper has 2425 boxes of 24 pencils each. How many pencils do all the boxes have in all?

In [43]:
total_boxe=2425
box_1=24
total_pencils=total_boxe * box_1
print(total_pencils)

58200


Question 3:- Linda bought a coat for $2265 and a saree for $2150. She gave $5000 to the shopkeeper. How much money did the shopkeeper return to her

In [44]:
coat_price=2265
saree_price=2150
total_money=5000
sum1=coat_price + saree_price
return_shop=total_money - sum1
print(return_shop)

585


Question 4:- The cost of 21 TV sets is $95844. Find the cost of one TV set.

In [47]:
total_tv= 21
price=95844
one_tv_cost =95844 / 21
print(one_tv_cost)

4564.0


Question 5:- A factory produces 24532 bulbs in a month. What is its annual production?

In [48]:
one_month_prod=24532
one_year_prod= one_month_prod * 12
print(one_year_prod)

294384


Question 6:- There are 145968 bags of sugar, 236487 bags of wheat and some bags of rice in a godown. It the total number of bags in the godown is 450000, find the number of bags of rice.

In [59]:
suger_bage=145968
wheat_bage=236487
total_bages=450000
sum=suger_bage + wheat_bage
rice_bag = total_bages - sum 
print(rice_bag,'bags')

67545 bags


Question 7:- A factory manufactured 483685 toys in three weeks. The production in first week was 146345 toys and in second week 138152 toys. Find the production in the third week.

In [60]:
three_weeks=483685
first_week=146345
second_week=138152
third_week=483685-(146345 + 138152)
print(third_week,'toys')

199188 toys


Question 8:- The cost of a sofa set is $9372. How much will 124 such sofa sets cost?

In [61]:
cost_sofa=9372
sofa_set_cost=124 * 9372
print(sofa_set_cost)

1162128


Question 9:- There are 86 rooms in a school. 4386 students study there. Equal number of students sits in each room?

In [62]:
total_rooms=86
total_student=4386
equality=total_student / total_rooms
print(equality,"students")

51.0 students


Question 10:- 1575 students of a school want to go Agra by bus. If one bus can carry 75 students, how many buses are required to carry all the students?

In [64]:
total_students=1575
one_bus=75
sub=total_students - one_bus
need_bus=sub /one_bus
print(need_bus,'buses')

20.0 buses


Question 11:- The cost of a radio set is 1475. What is the cost of 35 such radio sets?

In [65]:
one_radio=1475
total=one_radio * 35
print(total,'radios')

51625 radios


## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [4]:
cost_of_ice_bag=1.25
profit_margin=20
p=20/100
print(p)
bags_sells=500 
Profit_per_bag= p * cost_of_ice_bag 
total_profit=Profit_per_bag * bags_sells
print(total_profit)


0.2
125.0


Thus, the grocery store makes a total profit of $125. While this is a reasonable way to solve a problem, it's not entirely clear by looking at the code cell what the numbers represent. We can give names to each of the numbers by creating Python *variables*.

> **Variables**: While working with a programming language such as Python, information is stored in *variables*. You can think of variables as containers for storing data. The data stored within a variable is called its *value*.

If you try to view the value of a variable that has not been *defined*, i.e., given a value using the assignment statement `variable_name = value`, Python shows an error.

In [5]:
Profit margin = 20%

SyntaxError: invalid syntax (1293283398.py, line 1)

In [6]:
Profit margin = 20 value

SyntaxError: invalid syntax (1281338204.py, line 1)

Storing and manipulating data using appropriately named variables is a great way to explain what your code does.

Let's display the result of the word problem using a friendly message. We can do this using the `print` *function*.

> **Functions**: A function is a reusable set of instructions. It takes one or more inputs, performs certain operations, and often returns an output. Python provides many in-built functions like `print` and also allows us to define our own functions.

In [11]:
def totle_profit(cost_of_ice_bag,profit_margin,bags_sells):
    p=profit_margin/100
    Profit_per_bag= p * cost_of_ice_bag 
    total_profit=Profit_per_bag * bags_sells
    return total_profit
t=totle_profit(1.25,20,500)
print("totle profit is :",t)

 The `print` function is used to display information. It takes one or more inputs, which can be text (within quotes, e.g., `"this is some text"`), numbers, variables, mathematical expressions, etc. We'll learn more about variables & functions in the next tutorial.

Creating a code cell for each variable or mathematical operation can get tedious. Fortunately, Jupyter allows you to write multiple lines of code within a single code cell. The result of the last line of code within the cell is displayed as the output. 



In [12]:
print("Totle profit is :",t)

Totle profit is : 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [13]:
# Inline and single-line comments start with #
def totle_profit(cost_of_ice_bag,profit_margin,bags_sells):
#     profit_margin=20
    p=profit_margin/100
#     print(p)
    Profit_per_bag= p * cost_of_ice_bag 
    total_profit=Profit_per_bag * bags_sells
    return total_profit
t=totle_profit(1.25,20,500)
print("totle profit is :",t)

totle profit is : 125.0


In [14]:
#single-line comments start with """
def totle_profit(cost_of_ice_bag,profit_margin,bags_sells):
    """cost_of_ice_bag=1.25
    profit_margin=20
    bags_sells=500"""
    p=profit_margin/100
#     print(p)
    Profit_per_bag= p * cost_of_ice_bag 
    total_profit=Profit_per_bag * bags_sells
    return total_profit
t=totle_profit(1.25,20,500)
print("totle profit is :",t)

totle profit is : 125.0




> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [20]:
Flying_plane_costs = 5000
total_people=29
ticket_charges=200


In [21]:
total_earning=29*200
print(total_earning)
import numpy as np

5800


In [23]:
total_profit=Flying_plane_costs - total_earning
total_profit

-800

Python Absolute Value – abs() for real and complex numbers
The absolute value of a number refers to that value's magnitude, regardless of whether it is positive or negative. For example, the absolute value of -800 is 800.

In [24]:
print(abs(total_profit))

800


## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [1]:
# == Check if operands are equal
a=2
b=4
a ==b


False

In [5]:
c= 4
b == c

True

In [6]:
b== a+a

True

In [7]:
a== b/2

True

In [8]:
#let's try not equal operator != Check if operands are not equal
a!= b

True

In [9]:
a!=c

True

In [10]:
b!=a+a

False

In [11]:
a!=b-a

False

In [13]:
b!=b/a

True

In [14]:
# > Check if left operand is greater than right operand
a>b


False

In [15]:
b>a

True

In [16]:
b>a+a

False

In [ ]:
# < Check if left operand is less than right operand


In [17]:
a<b

True

In [18]:
b<a

False

In [19]:
a<b-a

False

In [20]:
b<b-a

False

In [21]:
a<b/a

False

In [ ]:
# >= Check if left operand is greater than or equal to right operand

In [22]:
b>=a

True

In [23]:
b>=a+a

True

In [24]:
a>=b/2

True

In [25]:
a>=b

False

In [26]:
a>=b-a

True

In [ ]:
# <= Check if left operand is less than or equal to right operand


In [27]:
a<=b

True

In [32]:
b<= b

True

In [29]:
b<=a++a

True

In [30]:
a<=b-a

True

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


In [33]:
# lets try The logical operators and
a=b=2
a == 2 and b==2

True

In [34]:
a==3 and b== 4

False

In [35]:
a==2 and b== 4/2

True

In [36]:
a== 4-2 and b== 2

True

In [37]:
a== 4 and b== 2

False

In [38]:
b== 2 and a== 4

False

In [39]:
a== 2 and b== 4

False

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |



In [ ]:
# lets try The or operator 

In [40]:
a ==2 or b== 2

True

In [41]:
a== 2 or b== 4

True

In [42]:
a== 4 or b== 2

True

In [43]:
a==4 or b==1

False

In [44]:
a== 1 or b==3

False

In [45]:
a== 2 or b==4

True

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [46]:
not a==2

False

In [51]:
not b==4 and not b==2

False

In [52]:
not b==2 or a==2

True

In [49]:
not a==5

True

In [50]:
not b==1

True

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

If parentheses are not used, logical operators are applied from left to right.

Experiment with arithmetic, conditional and logical operators in Python using the interactive nature of Jupyter notebooks. We will learn more about variables and functions in future tutorials.

There is also a ternary operator in Python. Learn about it here: https://data-flair.training/blogs/python-ternary-operator/

## Further Reading and References

Following are some resources where you can learn about more arithmetic, conditional and logical operations in Python:

* Python Tutorial at W3Schools: https://www.w3schools.com/python/
* Practical Python Programming: https://dabeaz-course.github.io/practical-python/Notes/Contents.html
* Python official documentation: https://docs.python.org/3/tutorial/index.html

Now that you have taken your first steps with Python, you are ready to move on to the next tutorial: ["A Quick Tour of Variables and Data Types in Python"](https://jovian.ml/aakashns/python-variables-and-data-types).

## Questions for Revision

Try answering the following questions to test your understanding of the topics covered in this notebook:

1. What is a Jupyter notebook? 
2. How do you add a new code cell below an existing cell?
3. How do you add a new Markdown cell below an existing cell?
4. How do you convert a code cell to a Markdown cell or vice versa?
5. How do you execute a code cell within Jupyter?
6. What the different arithmetic operations supported in Python?
7. How do you perform arithmetic operations using Python?
8. What is the difference between the `/` and the `//` operators?
9. What is the difference between the `*` and the `**` operators?
10. What is the order of precedence for arithmetic operators in Python?
11. How do you specify the order in which arithmetic operations are performed in an expression involving multiple operators?
12. How do you solve a multi-step arithmetic word problem using Python?
13. What are variables? Why are they useful?
14. How do you create a variable in Python?
15. What is the assignment operator in Python?
16. What are the rules for naming a variable in Python?
17. How do you view the value of a variable?
18. How do you store the result of an arithmetic expression in a variable?
19. What happens if you try to access a variable that has not been defined?
20. How do you display messages in Python?
21. What type of inputs can the print function accept?
22. What are code comments? How are they useful?
23. What are the different ways of creating comments in Python code?
24. What are the different comparison operations supported in Python?
25. What is the result of a comparison operation?
26. What is the difference between `=` and `==` in Python?
27. What are the logical operators supported in Python?
28. What is the difference between the `and` and `or` operators?
29. Can you use comparison and logical operators in the same expression?
30. What is the purpose of using parentheses in arithmetic or logical expressions?
31. What is Markdown? Why is it useful?
32. How do you create headings of different sizes using Markdown?
33. How do you create bulleted and numbered lists using Markdown?
34. How do you create bold or italic text using Markdown?
35. How do you include links & images within Markdown cells?
36. How do you include code blocks within Markdown cells?
37. Is it possible to execute the code blocks within Markdown cells?
38. How do you upload and share your Jupyter notebook online using Jovian?
39. What is the purpose of the API key requested by jovian.commit ? Where can you find the API key?
40. Where can you learn about arithmetic, conditional and logical operations in Python?


## Solution for Exercise



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [26]:
Flying_plane_costs = 5000
total_people=29
ticket_charges=200
total_earning=29*200
total_profit=Flying_plane_costs - total_earning
total_profit


-800

In [27]:
import numpy as np
print(abs(total_profit))

800


The travel company is in a profit of $800!